File: frontend/src/types/enums.ts
export const NodeType = {
    Service: "Service",
    Endpoint: "Endpoint",
    Deployment: "Deployment",
    Pod: "Pod",
    Node: "Node",
    Database: "Database",
    Table: "Table",
    QueueTopic: "QueueTopic",
    Cache: "Cache",
    ExternalAPI: "ExternalAPI",
    SecretConfig: "SecretConfig",
    Library: "Library",
    TeamOwner: "TeamOwner",
    SLASLO: "SLASLO",
    RegionCluster: "RegionCluster",
} as const;

export type NodeTypeValue = (typeof NodeType)[keyof typeof NodeType];

export const EdgeType = {
    Calls: "calls",
    PublishesTo: "publishesto",
    ConsumesFrom: "consumesfrom",
    Reads: "reads",
    Writes: "writes",
    DependsOn: "dependson",
    DeployedOn: "deployedon",
    OwnedBy: "ownedby",
    AuthenticatesVia: "authenticatesvia",
    RateLimitedBy: "ratelimitedby",
    FailsOverTo: "fails_over_to",
} as const;

export type EdgeTypeValue = (typeof EdgeType)[keyof typeof EdgeType];

export const LayoutAlgorithm = {
    Spring: "spring",
    KamadaKawai: "kamada_kawai",
    Circular: "circular",
    Shell: "shell",
} as const;

export type LayoutAlgorithmValue = (typeof LayoutAlgorithm)[keyof typeof LayoutAlgorithm];

export const ImpactDirection = {
    Upstream: "upstream",
    Downstream: "downstream",
    Both: "both",
} as const;

export type ImpactDirectionValue = (typeof ImpactDirection)[keyof typeof ImpactDirection];

File: app/models/enums.py
from enum import Enum


class NodeType(str, Enum):
    SERVICE = "Service"
    ENDPOINT = "Endpoint"
    DEPLOYMENT = "Deployment"
    POD = "Pod"
    NODE = "Node"
    DATABASE = "Database"
    TABLE = "Table"
    QUEUE_TOPIC = "QueueTopic"
    CACHE = "Cache"
    EXTERNAL_API = "ExternalAPI"
    SECRET_CONFIG = "SecretConfig"
    LIBRARY = "Library"
    TEAM_OWNER = "TeamOwner"
    SLA_SLO = "SLASLO"
    REGION_CLUSTER = "RegionCluster"


class EdgeType(str, Enum):
    CALLS = "calls"
    PUBLISHES_TO = "publishesto"
    CONSUMES_FROM = "consumesfrom"
    READS = "reads"
    WRITES = "writes"
    DEPENDS_ON = "dependson"
    DEPLOYED_ON = "deployedon"
    OWNED_BY = "ownedby"
    AUTHENTICATES_VIA = "authenticatesvia"
    RATE_LIMITED_BY = "ratelimitedby"
    FAILS_OVER_TO = "fails_over_to"
